test/bidir-full-sync-tests.js

```diff
@@ -47,6 +48,107 @@ test2('bidirectional full sync test', getClusterSizes(5), 20000,
 	})
 );
 
+test2('bidirectional full sync throttling test', getClusterSizes(3), 20000,
+    prepareCluster(function(t, tc, n) {
+        return [
+            dsl.assertStats(t, tc, n+1, 0, 0),
+            dsl.waitForEmptyPing(t, tc),
+
+            assertReverseFullSyncThrottling(t, tc, 0, n*3, 5),
+
+            dsl.assertStats(t, tc, n+1, 0, 0),
+        ];
+    })
+);
+
+// assert if the number of joins is throttled.
+function assertReverseFullSyncThrottling(t, tc, idx, count, expectedJoins) {
+    var functions = [];
+
+    //pause handling joins to stall reverse full syncs
+    functions.push(pauseHandlingJoins);
+    functions.push(dsl.waitForEmptyPing(t, tc));
+
+    // trigger the reverse full syncs
+    functions.push(triggerReverseFullSyncs(tc, idx, count));
+
+    // verify the number of join requests
+    functions.push(verify);
+
+    // handle the joins
+    functions.push(resumeHandlingJoins);
+
+    // the SUT sends a join request to perform a bidirectional full sync
+    functions.push(dsl.waitForJoins(t, tc, expectedJoins));
+
+    var joins = [];
+
+    return functions;
+
+    function triggerReverseFullSyncs(tc, nodeIx, count) {
+        var f = function triggerFullSync(list, cb) {
+            async.times(count, function triggerFullSync(i, next){
+                tc.fakeNodes[nodeIx].requestPing(next, undefined, {checksum: 1}); //override checksum to trigger a fullsync
+            }, function done() {
+                cb(list);
+            });
+        };
+        f.callerName = 'triggerReverseFullSyncs';
+
+        return f;
+    }
+
+    function pauseHandlingJoins(list, cb) {
```

Comment:
I like the name overwriteJoinHandlerWithPause or something similar better. Or some indication that we overwrite the join handler here